src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -562,6 +606,240 @@ void Gather::initShortParams(threadExecParams& p, const uint64_t start) {
     }
 }
 
+template <typename OUT_TYPE>
+void Gather::execWeightsCompressedU4() {
```

Comment:
can we have single execWeightsCompressed4Bit, but parametrize it with cvt_low and cvt_high functions. See this comment for reference: https://github.com/openvinotoolkit/openvino/pull/23287#discussion_r1538868144